Dockerfile

```diff
@@ -23,10 +23,10 @@ FROM git-dev AS git-prod
 WORKDIR /opt/kimai
 RUN rm -r tests
 
-# composer with prestissimo (faster deps install)
-FROM composer:1.10 AS composer
+# composer
+FROM composer:2.0 AS composer
 RUN mkdir /opt/kimai && \
-    composer --no-ansi require --working-dir=/opt/kimai hirak/prestissimo
+    composer --no-ansi require --working-dir=/opt/kimai
```

Comment:
Have you tested that? Before it was `composer require hirak/prestissimo` and now it is only a blank `composer require` without argument. Is that what we want to achieve in this step? What does happen in Composer 2 when a require is executed without further arguments?

Reply:
Without this line the build fails:

```sh
Step 46/80 : COPY --from=composer --chown=www-data:www-data /opt/kimai/vendor /opt/kimai/vendor
COPY failed: stat /var/lib/docker/btrfs/subvolumes/c938d7d8882b047ddbb066a11cd5a75904da67402565c43e2932c9be53695117/opt/kimai/vendor: no such file or directory
```

Reply:
What does happen in Composer 2 when a require is executed without further arguments?

```sh
Step 9/80 : RUN mkdir /opt/kimai &&     composer --no-ansi require --working-dir=/opt/kimai
 ---> Using cache
 ---> f3faab005812
```

```sh
Step 46/80 : COPY --from=composer --chown=www-data:www-data /opt/kimai/vendor /opt/kimai/vendor
 ---> Using cache
 ---> ea90032e2d11
```

Reply:
What do you mean "it works" - to me that looks like a cached result 😁 ?
See https://getcomposer.org/doc/03-cli.md#require - using require without an argument should trigger an interactive prompt (which might fail silently because of the docker env). 
And it might accidentally trigger an install - but I am against using the wrong command for the right thing

Reply:
Okay, now your answer changed - which is even better because you see that `./composer.json has been created`. 
Previously this might have been required because of the plugin. 
But I think this step should be refactored. For some reason the docker setup requires that this step is being executed, and I'd say: this is not correct. Not your fault, it was like that before... but we shouldn't rely on a weird/wrong behavior in the long run.

Reply:
I don't know the correct command to use, may be it should be [init](https://getcomposer.org/doc/03-cli.md#init) (`composer --no-ansi init --working-dir=/opt/kimai`) then?

```sh
Step 9/79 : RUN mkdir /opt/kimai &&     composer --no-ansi init --working-dir=/opt/kimai
 ---> Running in 0b133ec6aca7
Removing intermediate container 0b133ec6aca7
 ---> 31e446a38f77
```

Reply:
Using `init` fails as well:

```sh
Step 45/79 : COPY --from=composer --chown=www-data:www-data /opt/kimai/vendor /opt/kimai/vendor
COPY failed: stat /var/lib/docker/btrfs/subvolumes/b851a9c07c30e28652af1e1f47905efebf6acd212aab99944b9313f0cbae1312/opt/kimai/vendor: no such file or directory
```

Reply:
I think the problem is not that line (which from my view could be removed) - for me th problem is the copy command https://github.com/tobybatch/kimai2/blob/main/Dockerfile#L191
But I have no clue why it is there, needs to be answered by @tobybatch or @Schrolli91 probably.

Reply:
Instead of the `require`, we need the `install` command now...
Composer has to install all deps used from kimai. This will be copied from the composer build stage.

Reply:
We used a spearate build stage and installed the `hirak/prestissimo` here.
The compose.phar and the data from prestissimo was copied to the kimai build stage.
If prestissimo is no longer needed we can complete remove the line:
'RUN mkdir /opt/kimai && \
    composer --no-ansi require --working-dir=/opt/kimai hirak/prestissimo'

And only the composer.phar has to be copied, because there is no prestissimo anymore, so we can also remove:
`COPY --from=composer --chown=www-data:www-data /opt/kimai/vendor /opt/kimai/vendor`